### ISSUE TITLE
Clippy removing parenthesis from clap derive label makes the program fail to compile

### ISSUE THREAD

**Issue by Crowdedlight** [rust-lang/rust-clippy#13000]
### Summary

Creating report here as the console output wanted me to do that. Not sure if I am doing something different or its a known limitation/error. Or if this is counted as a false positive. 

Running `cargo clippy --fix` fails as it tries to fix the following code:

```rust
        #[clap(short, long, default_value_t=("templates.pdf".to_string()) )]
        output_path: String,
``` 
by suggesting removing the parenthesis so it becomes:
```rust
        #[clap(short, long, default_value_t="templates.pdf".to_string() )]
        output_path: String,
```
Which fails in the console with this output:
```
after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected `,`
  --> src/main.rs:26:60
   |
26 |         #[clap(short, long, default_value_t="templates.pdf".to_string() )]
   |                                                            ^

error: aborting due to 1 previous error
```

### Reproducer

I tried this code with clap. 

```rust
#[derive(Subcommand, Debug, Clone)]
enum Commands {
    Templates {
        #[clap(short, long, default_value_t=("templates.pdf".to_string()) )]
        output_path: String,
        #[arg(short, long, default_value_t=("templates.txt".to_string()) )]
        input: String
    },
    Ids {
        output_path: String,
        #[arg(short, long, default_value_t=1)]
        pages: u64,
    }
}
```

I expected to see this happen:
It would not remove the parenthesis in the ``default_value_t`` call as that seemingly break the program and fails to compile.   
 
Instead, this happened:
It removed them and the program fails to compile. 

### Version

```text
rustc 1.79.0 (129f3b996 2024-06-10)
binary: rustc
commit-hash: 129f3b9964af4d4a709d1383930ade12dfe7c081
commit-date: 2024-06-10
host: x86_64-unknown-linux-gnu
release: 1.79.0
LLVM version: 18.1.7
```

**Comment by Alexendoo** [comment]
Ah wait this is a rustc lint rather than a clippy one, looks like it's a known issue - https://github.com/rust-lang/rust/issues/104457, https://github.com/clap-rs/clap/issues/5119

Here I believe you could go with `default_value = "templates.pdf"`

**Comment by Crowdedlight** [comment]
> Ah wait this is a rustc lint rather than a clippy one, looks like it's a known issue - [rust-lang/rust#104457](https://github.com/rust-lang/rust/issues/104457), [clap-rs/clap#5119](https://github.com/clap-rs/clap/issues/5119)
> 
> Here I believe you could go with `default_value = "templates.pdf"`

Ahh, I didn't realise it was a rustc lint error. Thanks for pointing me in the right direction! Much appriciated!
